From zaitcev@redhat.com  Fri Jan 11 17:11:12 2008
From: Pete Zaitcev <zaitcev@redhat.com>
Date: Sat, 5 Jan 2008 02:01:07 -0800
Subject: USB: usbfs: struct async is a fixed size structure
To: greg@kroah.com
Cc: linux-usb@vger.kernel.org
Message-ID: <20080105020107.1f305cba.zaitcev@redhat.com>


The ISO descriptors are allocated separately in proc_submiturb for a fetch
from user mode, then tucked at the end of URB. This seems like a dead code.

Signed-off-by: Pete Zaitcev <zaitcev@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/devio.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/usb/core/devio.c
+++ b/drivers/usb/core/devio.c
@@ -210,9 +210,9 @@ err:
 
 static struct async *alloc_async(unsigned int numisoframes)
 {
-        unsigned int assize = sizeof(struct async) + numisoframes * sizeof(struct usb_iso_packet_descriptor);
-        struct async *as = kzalloc(assize, GFP_KERNEL);
+	struct async *as;
 
+	as = kzalloc(sizeof(struct async), GFP_KERNEL);
         if (!as)
                 return NULL;
 	as->urb = usb_alloc_urb(numisoframes, GFP_KERNEL);
